libid/ui/make_batch_file.cpp

```diff
@@ -828,6 +829,16 @@ static void write_batch_params(const char *color_inf, bool colors_only, int max_
             put_param(" passes=%c%c", g_user_std_calc_mode, (char)g_stop_pass + '0');
         }
 
+        if (g_perturbation != PerturbationMode::AUTO)
+        {
+            put_param(" %s=%s", "perturbation", (g_perturbation == PerturbationMode::YES) ? "yes" : "no");
```

Comment:
Don't use %s for a string literal, just write the text as part of the format string, e.g. `put_param(" perturbation=%s", ...`

Needs to handle optional tolerance here